src/System.Net.Sockets/src/System/Net/Sockets/IOControlKeepAlive.Windows.cs

```diff
@@ -39,6 +48,11 @@ public static int Get(SafeSocketHandle handle, SocketOptionName optionName)
                 MillisecondsToSeconds(ioControlKeepAlive?._intervalMs ?? WindowsDefaultIntervalMs);
         }
 
+        public static SocketError Set(SafeSocketHandle handle, SocketOptionName optionName, byte[] optionValueSeconds)
+        {
+            return Set(handle, optionName, BitConverter.ToInt32(optionValueSeconds, 0));
```

Comment:
ToInt32 will throw an ArgumentOutOfRangeException if optionValueSeconds is too short.  We should check its length first and return the right SocketError (e.g. whatever SocketError Windows returns when we're not emulating this) if it's too short, so that ToInt32 won't throw.  And we should have a test for that :)

Reply:
Similarly we need to check the array for null and return the right error.

Reply:
`SocketError.Fault` needs to be used since it is what generates an exception for:
```csharp
byte[] bufferNull = null;
SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveBuffer, bufferNull);

byte[] bufferLenghtZero = new byte[0];
SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveBuffer, bufferLenghtZero);

byte[] bufferShort = new byte[1];
SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveBuffer, bufferShort);
```